app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/provision_workflow/general.rb

```diff
@@ -6,7 +6,10 @@ module ManageIQ::Providers::IbmCloud::VPC::CloudManager::ProvisionWorkflow::Gene
   # @param _options [void]
   # @return [Hash] Hash with ems_ref as key and name as value.
   def provision_type_to_profile(_options = {})
-    @provision_type_to_profile ||= index_dropdown(ar_ems.flavors)
+    @provision_type_to_profile ||= begin
+      flavors = resources_for_ui[:ems] ? ar_ems.flavors : ManageIQ::Providers::IbmCloud::VPC::CloudManager::Flavor.all
+      index_dropdown(flavors)
+    end
```

Comment:
```suggestion
    flavors = resources_for_ui[:ems] ? ar_ems.flavors : ManageIQ::Providers::IbmCloud::VPC::CloudManager::Flavor.all
    @provision_type_to_profile = index_dropdown(flavors)
```

Reply:
(Similar pattern can be applied to the rest of the places)

Reply:
Hm I don't know about that one, that would kind of defeat the purpose of memoizing the result since it'd issue the query everytime

Maybe
```ruby
def provision_type_to_profile
  return @provision_type_to_profile if @provision_type_to_profile.present?

  flavors = resources_for_ui[:ems] ? ar_ems.flavors : ManageIQ::Providers::IbmCloud::VPC::CloudManager::Flavor.all
  @provision_type_to_profile = index_dropdown(flavors)
end
```

But I'm not sure that's drastically better than what I have already?